### PR TITLE
Feature: show providers' public and private documents in the attachment manager

### PR DESCRIPTION
**Pull request by LiamStanziani**
## Summary

Adds provider-owned documents to the attachment manager so providers can attach their own private or public eDocs to consultations and eForms, alongside the existing patient-specific documents.

## User-facing changes

- Attachment manager now shows three document sections:
  - **Patient Documents** (renamed from "Documents")
  - **Public eDocs** — shared provider library
  - **Private eDocs** — current provider's private library
- Private doc selections trigger a confirmation dialog warning that private documents will become visible to anyone with access to the record once attached. The warning only fires on *new* attachments per the spec.
- Cross-provider private-doc attachments render in italics with an `(attached by another provider)` label, so reviewing providers can see that a doc they're looking at belongs to a different provider's library.
- Soft-deleted attached docs remain retrievable and render in italics with a `(deleted)` label so they don't silently vanish from saved consults/eForms after a library cleanup.

### New consultation attachment page:

image

### Cross-provider consultation attachment page (deleted documents, attached documents by other providers):

image

### Warning popup on attachment page when attaching new private provider documents

image

**Note:** since both consultation and eForm's use the same base JSP for the attachment manager, I have only shown screenshots based on the consultation flow since attaching both shouldn't show any differences.

### DAO layer — unchanged shape

The 3 attached-doc DAO methods (`findDocsAndConsultDocsByConsultId`, `findDocsAndEFormDocsByFdid`, `findDocsAndConsultResponseDocsByConsultId`) are unchanged versus develop. Their pre-PR 2-column contract is preserved so the 16 downstream callers (PDF printing, fax, REST APIs, etc.) see no behavioral change.

### EDocUtil — ctl enrichment for attached docs

New `enrichAttachedWithCtl` helper runs after the DAO returns attached-doc tuples. It does one batch query via a new `CtlDocumentDao.findByDocumentNos(List<Integer>)` to fetch ctl rows for all attached docs and picks the preferred ctl per doc (provider bindings win over demographic so library docs classify correctly). Orphan docs with no ctl row pass through unclassified and fall through to the patient section.

### DocumentPreview2Action — attached-state classification

New `mergeAttachedContext` / `mergeSingleAttachedDoc` walk the attached-doc list and classify each into the correct section (patient / provider public / provider private), re-injecting soft-deleted items and flagging cross-provider private docs via `foreignPrivateDocIds`.

### JSP and JS

- Three new sections in `attachDocument.jsp` with select-all + show-more + preview + deleted/foreign markers.
- Shared helpers in the JSP `<script>` block: `confirmPrivateDocsIfAny`, `buildDelegateInput`, `syncPreCheckedToDelegates` — consumed by both the consult JSP and the eForm toolbar.
- `data-delegate-type="doc"` attribute on doc delegate inputs so the toolbar JS resolves checkboxes by name+value across the three sections (a plain `#docNo<id>` selector would only match the patient section).
- Dialog reopen sync: unchecks any server-pre-checked box whose delegate was removed in a prior dialog session (fixes the issue where an unchecked doc came back checked on reopen before the consult/eForm was saved).

## Privacy and PHI considerations

The attachment uses a **reference-based** model: attaching a provider's private doc to a consult creates a row in `consultdocs` (or `eform_docs`) pointing at the original `document_no`, not a copy.

Key property: there is **no reverse lookup** in the application code that takes a `document_no` and returns every patient/consult/eForm it's attached to. Each forward lookup (`DisplayDemographicConsultationRequests`, eForm display, etc.) is scoped to a single request/fdid, so one provider document attached to multiple patients' records never becomes cross-patient visibility through app code.

### Automated

- `DocumentDaoAttachedFindIntegrationTest` — 14 cases across 3 nested classes covering each attached-doc query: happy path, multi-doc, orphan docs (no ctl row), soft-deleted, scope-by-id, non-DOC docType exclusion.
- `CtlDocumentDaoFindByDocumentNosIntegrationTest` — 6 cases for the new batch method: null input, empty input, single doc, multiple docs, doc with multiple ctl rows, scope-by-doc-ids.
- `DocumentPreview2ActionMergeUnitTest` — 12 cases across 5 nested classes covering the full matrix of `mergeSingleAttachedDoc` classification: patient / public provider / own private / foreign private × active / deleted, plus edge cases (null current provider, legacy "provider" vs "providers" module spelling).

### Manual smoke tests performed

Covered all scenarios from the spec's "Testing Performed" section plus several derived ones:

- Fresh attachment dialog renders all three sections ✓
- Private-doc attach triggers warning; public-doc attach does not ✓
- Warning does NOT fire when reopening a saved consult/eForm with pre-existing attachments and no changes (spec's no-change exemption) ✓
- Dialog reopen sync: uncheck → close → reopen → stays unchecked ✓
- Each provider only sees their own private docs in the picker (Provider B cannot browse Provider A's library) ✓
- Provider B sees Provider A's attached private docs with "(attached by another provider)" label ✓
- Soft-deleted attached docs remain retrievable with "(deleted)" label ✓
- eForm save → reopen → provider-section boxes still pre-checked ✓
- eForm Email workflow includes attached provider docs in the PDF ✓

### Playwright verification

Confirmed the following at the DOM level in a running dev instance:

- Dialog renders Patient Documents, Public eDocs, Private eDocs with correct items and CSS classes (`providerPrivateDoc`, `foreign-doc`, `deleted-doc`).
- `syncPreCheckedToDelegates` correctly unchecks/reverts class/strips `data-pre-attached` when delegate is missing, and leaves state alone when delegate is present.
- Cross-provider scenario on consult 106: all 6 of openodoc's private docs rendered under Private eDocs with `foreign-doc` class and "(attached by another provider)" label.

## Design decisions worth noting

- **Reference vs. copy** — reference for both eForm and consult. The spec flags this as a reviewable design question; per-workflow split ("copy for consult, reference for eForm") can be added in a follow-up once user feedback is gathered.
- **Provider-wins ctl tiebreaker** — in the hypothetical case where a doc has multiple ctl bindings, the provider binding wins so library docs classify consistently. In the current data pattern this doesn't occur (invariant: one ctl per `document_no`, verified via `GROUP BY document_no HAVING COUNT(*) > 1` → 0 rows).
- **Warning exemption on no-change** — reopening a dialog on a pre-existing private attachment and making no changes does not re-trigger the warning (matches spec).

## Summary by Sourcery

Expose provider-owned public and private eDocs alongside patient documents in the attachment manager, while keeping attachments secure and accurately classified across consult and eForm workflows.

New Features:
- Add public and private provider eDoc sections to the attachment dialog for consultations and eForms, with select-all, pagination, and preview support.
- Display visual markers and labels for deleted documents and private documents owned by other providers in the attachment manager and consuming UIs.

Bug Fixes:
- Ensure attachment selections persist correctly across dialog reopenings and that unchecked pre-attached items remain unchecked.
- Fix validation so Ocean and other workflows accept attachments that come from provider libraries as well as patient-scoped documents.

Enhancements:
- Classify attached documents into patient, provider public, and provider private sections using ctl_document data while preserving existing DAO contracts.
- Harden parameter parsing and security checks when fetching attachable documents, including explicit _edoc privilege checks.
- Escape various document- and form-derived strings in the attachment JSP to reduce XSS risk.

Tests:
- Add integration tests for DocumentDao attached-document queries and for validating attachable document IDs against demographic and provider-library scope.
- Add integration tests for CtlDocumentDao batch lookups used to enrich attached documents.
- Add unit tests covering mergeAttachedIntoSections classification logic across patient/public/private and ownership/deleted permutations.

### CODE REVIEW

**Comment by LiamStanziani** [comment]
Alright so the diff count seems to be different between the develop and the release version of these changes, some more information below on this:

---

Overall: +1666 (release PR) vs +1625 (develop PR) diffs

1666 - 1625 = 41

+41 diffs between these both

src/main/java/ca/openosp/openo/documentManager/actions/DocumentPreview2Action.java

On file: +94 (release PR) vs +53 (develop PR) diffs

94 - 53 = 41

ALL increased + diffs seem to be coming from this file, seems like for some reason the javadoc comments on methods `fetchConsultDocuments` and `fetchEFormDocuments` were not merging correctly in the release version, so it had to write the javadoc comments fully (develop seemed to already have these comments, they were just modified in the branch)

File: src/main/webapp/documentManager/attachDocument.jsp [hunk 1]
```diff
@@ -287,6 +294,65 @@
...
+         * parents can call them from their beforeClose handlers without needing
+         * a separate script include. Safe because the dialog is guaranteed loaded
+         * when beforeClose fires.
+         */
+
+        /** Warn on new (not pre-attached) private eDoc selections; returns false to abort the close. */
+        function confirmPrivateDocsIfAny(formSelector) {
+            var newPrivate = jQuery(formSelector)
+                .find(".providerPrivateDocument_check:checked:not(input[disabled='disabled'])")
+                .not('[data-pre-attached="true"]');
```
[the remark this answers is not in the document]
Comment (LiamStanziani) [reply src/main/webapp/documentManager/attachDocument.jsp:309]: This is fine, it works currently
